packages/flutter_tools/lib/src/commands/build_aot.dart

```diff
@@ -0,0 +1,129 @@
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import '../base/process.dart';
+import '../build_configuration.dart';
+import '../globals.dart';
+import '../runner/flutter_command.dart';
+import '../toolchain.dart';
+import 'run.dart';
+
+const String _kDefaultAotOutputDir = 'build/aot';
+
+class BuildAotCommand extends FlutterCommand {
+  BuildAotCommand() {
+    usesTargetOption();
+    addBuildModeFlags();
+    usesPubOption();
+    argParser.addOption('output-dir', defaultsTo: _kDefaultAotOutputDir);
+  }
+
+  @override
+  final String name = 'aot';
+
+  @override
+  final String description = "Build an ahead-of-time compiled snapshot of your app's Dart code.";
+
+  @override
+  Future<int> runInProject() async {
+    String outputPath = buildAotSnapshot(
+      findMainDartFile(argResults['target']),
+      outputPath: argResults['output-dir']
+    );
+    if (outputPath == null)
+      return 1;
+
+    printStatus('Built $outputPath');
+    return 0;
+  }
+}
+
+String _getSdkExtensionPath(String packagesPath, String package) {
+  Directory packageDir = new Directory(path.join(packagesPath, package));
+  return path.join(path.dirname(packageDir.resolveSymbolicLinksSync()), 'sdk_ext');
+}
+
+String buildAotSnapshot(
+  String mainPath, {
+  String outputPath: _kDefaultAotOutputDir
+}) {
+  String engineSrc = ToolConfiguration.instance.engineSrcPath;
+  if (engineSrc == null) {
+    printError('AOT compilation requires --engine-src-path');
+    return null;
+  }
+
+  String engineOut = ToolConfiguration.instance.getEngineArtifactsDirectory(
+      TargetPlatform.android_arm, BuildMode.profile).path;
+  String genSnapshot = path.join(engineOut, 'clang_x86', 'gen_snapshot');
+
+  Directory outputDir = new Directory(outputPath);
+  outputDir.createSync(recursive: true);
+  String vmIsolateSnapshot = path.join(outputDir.path, 'snapshot_aot_vmisolate');
+  String isolateSnapshot = path.join(outputDir.path, 'snapshot_aot_isolate');
+  String instructionsBlob = path.join(outputDir.path, 'snapshot_aot_instr');
+  String rodataBlob = path.join(outputDir.path, 'snapshot_aot_rodata');
+
+  String bindingsSrc = path.join(engineSrc, 'sky', 'engine', 'bindings');
+  String vmEntryPoints = path.join(bindingsSrc, 'dart_vm_entry_points.txt');
+  String vmEntryPointsAndroid = path.join(bindingsSrc, 'dart_vm_entry_points_android.txt');
+
+  String packagesPath = path.absolute(Directory.current.path, 'packages');
+  if (!FileSystemEntity.isDirectorySync(packagesPath)) {
+    printError('Could not find packages directory: $packagesPath\n' +
+               'Did you run `pub get` in this directory?');
+    return null;
+  }
+
+  String mojoSdkExt = _getSdkExtensionPath(packagesPath, 'mojo');
+  String mojoInternalPath = path.join(mojoSdkExt, 'internal.dart');
+
+  String skyEngineSdkExt = _getSdkExtensionPath(packagesPath, 'sky_engine');
```

Comment:
> I don't know if there is a way to discover these mappings dynamically.

Yes, I couldn't think of one either :)

Without this logic being data-driven, or a test running continuously, this will bit rot w/o us knowing it. Can you open an issue to run `build aot` on the chromium bots? It's already doing a local engine build, and has steps that test the other flutter commands.